apps/src/sites/studio/pages/blocks/edit.js

```diff
@@ -70,36 +72,65 @@ function initializeEditPage(defaultSprites) {
 
   if (blocks) {
     updateBlockPreview();
+    validateBlockConfig();
   }
+  setSubmitButtonState();
 
   $('.alert.alert-success')
     .delay(5000)
     .fadeOut(1000);
 }
 
 function onUpdateLinting(_, errors) {
-  const submitButton = document.querySelector('#block_submit');
   if (errors.length) {
-    submitButton.setAttribute('disabled', 'disabled');
+    hasLintingErrors = true;
   } else {
-    submitButton.removeAttribute('disabled');
+    hasLintingErrors = false;
   }
+  setSubmitButtonState();
 }
 
+const setSubmitButtonState = () => {
+  const disabled = hasLintingErrors || !isValidBlockConfig;
+  ['#block_clone', '#block_submit'].forEach(buttonId => {
+    const button = document.querySelector(buttonId);
+    if (disabled) {
+      button.setAttribute('disabled', 'disabled');
+    } else {
+      button.removeAttribute('disabled');
+    }
+  });
+};
+
 function validateBlockConfig(editor) {
   try {
     if (editor) {
       JSON.parse(editor.getValue());
     }
     updateBlockPreview();
+    validateBlockRenders();
+    isValidBlockConfig = true;
     validationDiv.text('Config and helper code appear valid.');
     validationDiv.css('color', VALID_COLOR);
   } catch (err) {
+    isValidBlockConfig = false;
     validationDiv.text(err.toString());
     validationDiv.css('color', INVALID_COLOR);
   }
 }
 
+// Validation only relevant for blocks rendered via google blockly
+// "unknown block" is only implemented in google blockly
+function validateBlockRenders() {
+  if (
+    Blockly.mainBlockSpace
+      .getAllBlocks()
+      .some(block => block.getFieldValue('NAME')?.includes('unknown block'))
+  ) {
+    throw 'Unable to render block with given configuration.';
```

Comment:
Perhaps add to the msg a brief explanation of why unable to render for the levelbuilder so they understand it's not a syntax error. Something like: 
'Unable to render block with given configuration because only Google Blockly blocks are valid.'

Reply:
Unknown blocks are created for any block that we can't deserialize. We basically just catch any errors in `domToBlockHeadless_` and instead create an unknown block so the page doesn't crash. The block itself won't have additional context about why it couldn't be rendered as configured.

see: https://github.com/code-dot-org/code-dot-org/pull/43989

Reply:
Thanks for the explanation!

Reply:
I do agree that the error message isn't great, let me see if I can add a little more detail.

Reply:
Updated error message to "Blockly is unable to render a block with the given configuration.", LMK what you think!